### DISCUSSION TITLE
how to get value of usestate

### DISCUSSION THREAD

**Discussion by Chrysanthemum536** [nuxt/nuxt#17470] [Questions] [answered]
how to get value of usestate?
i put
`<script setup>
  const a=useState("a");
  console.log(a);
</script>`
but it showed some weird object instead of the correct value of a

**Comment by Diizzayy** [comment, marked as answer]
@aqtsalt `useState` [returns a Ref](https://vuejs.org/api/reactivity-core.html#ref), you have to use the `.value` property to access it's value.

```vue
<script setup>
const a = useState("a");

// update ref value
a.value = "hello";

console.log(a.value);
</script>
```

I suggest you have a look at the [Vue 3 Reactivity API documentation](https://vuejs.org/api/reactivity-core.html#reactivity-api-core) to get a better understanding
